### PR TITLE
Feature/99 colo ddp integration

### LINKED ISSUE #99: fix _FullyShardedDataParallelMapping when running test_fsdp.py
## How to reproduce

```
python -m torch.distributed.run --nproc_per_node=2 --master_port=2333 ./tests/torch/nn/parallel/data_parallel/test_fsdp.py
```

## Environment

- OS :
- Python version :
- Transformers version :
- Whether to use Docker:
- Misc.:

The problem is in _fsdp where _FullyShardedDataParallelMappingForHuggingFace is used instead of _FullyShardedDataParallelMapping 
`
from oslo.transformers.mapping_utils import (
    _FullyShardedDataParallelMappingForHuggingFace,
)
`

### PR DESCRIPTION
**Pull request by dongsungkim**
## Title

- Initial coloDDP integration
- ColoDDP name will be changed later
- layer folder for zeroDDP will be added separately

## Description
- Integrate coloDDP for patricstar and zeroDDP

## Linked Issues

- resolved #99

### CODE REVIEW

**Comment by dongsungkim** [comment]
pytroch native DDP : 7.67764 vs coloDDP

**Comment by hyunwoongko** [comment]
image

File: oslo/torch/nn/parallel/data_parallel/_coloddp/data_parallel.py [hunk 1]
```diff
@@ -0,0 +1,589 @@
...
+        module (torch.nn.Module): Module to apply DDP.
+        process_group (Optional[dist.ProcessGroup], optional): The process group which DDP uses.
+            If it's None, the default data parallel group will be used. Defaults to None.
+        parallel_context : replace colossal ai's process_group funcitionality
+    """
+
+    def __init__(
+        self,
+        module: torch.nn.Module,
+        process_group: Optional[
```
Comment (hyunwoongko) [inline oslo/torch/nn/parallel/data_parallel/_coloddp/data_parallel.py:72]: I think process_group is unnecessary if there is parallel_context.

File: oslo/torch/nn/parallel/data_parallel/_coloddp/data_parallel.py [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+    if isinstance(args, torch.Tensor) and torch.is_floating_point(args):
+        args = args.to(dtype)
+    elif isinstance(args, (list, tuple)):
+        args = type(args)(_cast_float(t, dtype) for t in args)
+    elif isinstance(args, dict):
+        args = {k: _cast_float(v, dtype) for k, v in args.items()}
+    return args
+
+
+class ColoDDP(torch.nn.Module):
```
Comment (hyunwoongko) [inline oslo/torch/nn/parallel/data_parallel/_coloddp/data_parallel.py:49]: Let's use DistributedDataParallel as name.
remove all the `colo`.

**Comment by hyunwoongko** [comment]
and please run `precommit run --all-files`

File: oslo/torch/nn/parallel/data_parallel/_coloddp/data_parallel.py (hunk 2 above)
Comment (dongsungkim) [reply oslo/torch/nn/parallel/data_parallel/_coloddp/data_parallel.py]: Done

**Comment by dongsungkim** [comment]
current it has _ddp and _coloddp because we need to have some tests for TP and others before the change.
This test will be followed soon

**Pull request merged**